### PR TITLE
[PF-1070] Pass in location to ComputeModal so we don't constantly call GCS API

### PR DESCRIPTION
**Pull request by wnojopra**
This passes in location as a prop to ComputeModal. We move the ajax call to Analyses, NotebookLauncher, AppLauncher, and CloudEnvironmentLauncher.

I tested manually by going to the pages that the launchers are on (like /workspaces/:namespace/:name/analysis/launch/:analysisName)

### CODE REVIEW

**Comment by jdcanas** [comment]
A quick search of the code shows a few more places where ComputeModal is used, `RuntimeManager` and `AnalysisLauncher` in particular. Any reason it's not included in those places? 

Also, it seems it's included in `CloudEnvironmentModal` as a param, but never passed into the callsite of that component in `ContextBar`.

I haven't had time to test this yet, but I will shortly.

**Comment by wnojopra** [comment]
> a few more places where ComputeModal is used, RuntimeManager and AnalysisLauncher in particular.

I believe RuntimeManager is passing in location to ComputeModal in this PR, but let me know if I'm missing something.
You're right about AnalysisLauncher. I mentally mixed that with AnalysisModal and didn't update the Launcher.

> Also, it seems it's included in CloudEnvironmentModal as a param, but never passed into the callsite of that component in ContextBar.

I believe its correct in this PR. I merged a different PR yesterday that introduces location to ContextBar. This PR passes it through to CloudEnvironmentModal.

File: src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ export const CloudEnvironmentModal = ({
     tool,
```
Comment (kyuksel) [inline src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js:58]: Not something changed by your PR but I noticed that `onSuccess` and `bucketName` (on line 41 and 42, respectively) don't seem to be used.

File: src/pages/workspaces/workspace/applications/AppLauncher.js [hunk 2]
```diff
@@ -71,6 +72,19 @@ const ApplicationLauncher = _.flow(
...
+    )(async () => {
+      const { location } = await Ajax()
+        .Workspaces
+        .workspace(workspace.namespace, workspace.name)
+        .checkBucketLocation(googleProject, bucketName)
+      setLocation(location)
+    })
+
+
+    loadBucketLocation()
```
Comment (kyuksel) [inline src/pages/workspaces/workspace/applications/AppLauncher.js]: Would it be easier to read if one of the blank lines before `loadBucketLocation()` were moved to after it?

```suggestion

    loadBucketLocation()

```

File: src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js [hunk 3]
```diff
@@ -55,9 +55,24 @@ const AnalysisLauncher = _.flow(
     const { runtimeName, labels } = runtime || {}
     const status = getConvertedRuntimeStatus(runtime)
     const [busy, setBusy] = useState()
+    const [location, setLocation] = useState(defaultLocation)
     const { mode } = queryParams
     const toolLabel = getTool(analysisName)
     const iframeStyles = { height: '100%', width: '100%' }
```
Comment (kyuksel) [inline src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js:61]: Would it be easier to notice the `useEffect` block with below?

```suggestion

    const iframeStyles = { height: '100%', width: '100%' }
```

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js [hunk 4]
```diff
@@ -42,14 +42,29 @@ const NotebookLauncher = _.flow(
...
-  ({ queryParams, notebookName, workspace, workspace: { workspace: { namespace, name }, accessLevel, canCompute }, runtimes, persistentDisks, refreshRuntimes },
+  ({ queryParams, notebookName, workspace, workspace: { workspace: { bucketName, googleProject, namespace, name }, accessLevel, canCompute }, runtimes, persistentDisks, refreshRuntimes },
     ref) => {
     const [createOpen, setCreateOpen] = useState(false)
     const runtime = getCurrentRuntime(runtimes)
     const { runtimeName, labels } = runtime || {}
     const status = getConvertedRuntimeStatus(runtime)
     const [busy, setBusy] = useState()
     const { mode } = queryParams
+    const [location, setLocation] = useState(defaultLocation)
```
Comment (kyuksel) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: We usually cluster together `state`s. Not sure why we didn't in this file but it'd be nice to start:

```suggestion
    const [createOpen, setCreateOpen] = useState(false)
    const [busy, setBusy] = useState()
    const [location, setLocation] = useState(defaultLocation)

    const runtime = getCurrentRuntime(runtimes)
    const { runtimeName, labels } = runtime || {}
    const status = getConvertedRuntimeStatus(runtime)
    const { mode } = queryParams

```

**Review by kyuksel — COMMENTED** [review]
The code looks good to me with a few minor comments inline.
Testing so far looks good too but I need more time for further testing.

**Comment by kyuksel** [comment]
I've tested the following on both `Notebooks` and `Analyses` tabs so far with no issues:
- Jupyter runtime
  - Terminal
  - Spark UI
- RStudio runtime

File: src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js (hunk 1 above)
Comment (wnojopra) [reply src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js:58]: Fixed.

File: src/pages/workspaces/workspace/applications/AppLauncher.js (hunk 2 above)
Comment (wnojopra) [reply src/pages/workspaces/workspace/applications/AppLauncher.js]: Yes! Fixed.

File: src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js (hunk 3 above)
Comment (wnojopra) [reply src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js:61]: Yes, fixed.

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js (hunk 4 above)
Comment (wnojopra) [reply src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: Done.

**Review by kyuksel — APPROVED** [review]
👍🏼 pending integration tests passing

**Review by jdcanas — APPROVED** [review]
lgtm

**Pull request merged**